lib/grape-swagger.rb

```diff
@@ -468,12 +468,28 @@ def setup(options)
 
                     models = @@documentation_class.models_with_included_presenters(models.flatten.compact)
 
+                    modified_route_params = {}
+                    array_param = nil
+                    route.route_params.each_key do |k|
+                      if route.route_params[k].is_a?(Hash) && route.route_params[k][:type] == 'Array'
+                        array_param = k
+                      else
+                        new_key = k
+                        unless array_param.nil?
+                          if k.to_s.start_with?(array_param.to_s + '[')
+                            new_key = array_param.to_s + '[]' + k.to_s.split(array_param)[1]
+                          end
+                        end
+                        modified_route_params[new_key] = route.route_params[k]
+                      end
+                    end
```

Comment:
You should refactor this away out of here. Or maybe even better, why isn't this invoked from  `parse_params`?